### PR TITLE
controller: add controller key to log metadata

### PR DESCRIPTION
**Pull request by kopiczko**
Fixes giantswarm/giantswarm#4971.

This also brings improvements to setting logger metadata:

- Sets keys very early in the loop.
- Almost all keys are set in single method of the Controller object.

### CODE REVIEW

File: controller/controller.go [hunk 1]
```diff
@@ -219,16 +219,57 @@ func (c *Controller) DeleteFunc(obj interface{}) {
 // ProcessEvents takes the event channels created by the operatorkit informer
 // and executes the controller's event functions accordingly.
 func (c *Controller) ProcessEvents(ctx context.Context, deleteChan chan watch.Event, updateChan chan watch.Event, errChan chan error) error {
+	loop := -1
+
 	operation := func() error {
 		for {
+			loop++
```
Comment (rossf7) [inline controller/controller.go]: Maybe dumb question but what is the loop variable for? Is it added to the context for debugging?

Comment (rossf7) [reply controller/controller.go]: OK that makes sense. Filtering the logs for the loop is nice 👍.

File: controller/controller.go [hunk 2]
```diff
@@ -284,17 +327,8 @@ func (c *Controller) UpdateFunc(oldObj, newObj interface{}) {
...
-		if !ok {
-			meta = loggermeta.New()
-		}
-		meta.KeyVals["event"] = "update"
-
-		ctx = loggermeta.NewContext(ctx, meta)
-	}
-
 	ok, err := c.addFinalizer(ctx, obj)
+
```
Comment (kopiczko) [inline controller/controller.go]:
```suggestion
```

**Comment by kopiczko** [comment]
Looks good:

```
D 12/28 11:19:28 finalizer-integration-reconciliation-test/nodeconfigs/test-obj stop reconciliation due to finalizer added | operatorkit/controller/controller.go:341 | controller=test-operator | event=update | loop=0 | version=369
E 12/28 11:19:39 finalizer-integration-reconciliation-test/nodeconfigs/test-obj test-resource stop reconciliation due to error | operatorkit/controller/controller.go:208 | controller=test-operator | event=delete | loop=3 | version=377
	/go/src/github.com/giantswarm/operatorkit/controller/controller.go:462: 
	/go/src/github.com/giantswarm/operatorkit/controller/integration/testresource/testresource.go:72: 
	/go/src/github.com/giantswarm/operatorkit/controller/integration/test/controlflow/controlflow_test.go:149: 
	test error
E 12/28 11:19:50 finalizer-integration-reconciliation-test/nodeconfigs/test-obj test-resource stop reconciliation due to error | operatorkit/controller/controller.go:208 | controller=test-operator | event=delete | loop=4 | version=377
	/go/src/github.com/giantswarm/operatorkit/controller/controller.go:462: 
	/go/src/github.com/giantswarm/operatorkit/controller/integration/testresource/testresource.go:72: 
	/go/src/github.com/giantswarm/operatorkit/controller/integration/test/controlflow/controlflow_test.go:149: 
	test error
D 12/28 11:20:01 finalizer-integration-reconciliation-test/nodeconfigs/test-obj test-resource removing finalizer 'operatorkit.giantswarm.io/test-operator' | operatorkit/controller/finalizer.go:127 | controller=test-operator | event=delete | loop=5 | version=377
D 12/28 11:20:01 finalizer-integration-reconciliation-test/nodeconfigs/test-obj test-resource removed finalizer 'operatorkit.giantswarm.io/test-operator' | operatorkit/controller/finalizer.go:168 | controller=test-operator | event=delete | loop=5 | version=377
D 12/28 11:20:01 finalizer-integration-reconciliation-test/nodeconfigs/test-obj test-resource removing finalizer 'operatorkit.giantswarm.io/test-operator' | operatorkit/controller/finalizer.go:127 | controller=test-operator | event=delete | loop=6 | version=392
D 12/28 11:20:01 finalizer-integration-reconciliation-test/nodeconfigs/test-obj test-resource removed finalizer 'operatorkit.giantswarm.io/test-operator' | operatorkit/controller/finalizer.go:168 | controller=test-operator | event=delete | loop=6 | version=392
```

Interesting is where is `loop=1` and `loop=2`.

**Pull request merged**